frontend/src/Settings/settingsReducer.js

```diff
@@ -0,0 +1,30 @@
+import {
+  FETCH_SETTINGS_REQUEST, FETCH_SETTINGS_SUCCESS, FETCH_SETTINGS_FAILURE,
+} from './actionCreator';
+
+const initialState = {
+  email: 'example@email.com',
+  username: 'example name',
+  kingdomName: 'your kingdom\'s name',
+  isLoading: false,
+  error: undefined,
+};
+
+export default function settings(state = initialState, action) {
+  switch (action.type) {
+    case FETCH_SETTINGS_REQUEST:
+      return { ...state, isLoading: true };
+    case FETCH_SETTINGS_SUCCESS:
+      return {
+        ...state,
+        email: action.payload.email,
+        username: action.payload.username,
+        kingdomName: action.payload.kingdomName,
```

Comment:
what happens if the user changes only the email address? the `action.payload.username` for example will be undefined?

Reply:
This is an action to fetch the current setting values. Since we decided not to implement it, it will be replaced by an action posts the settings to the back end. Can I still name it fetchSettings or should it be saveSettings/postSettings?

Reply:
Also, since the front end no long stores the current setting values, do we still need redux for this? Can it just be a fetch in the component?